### DISCUSSION TITLE
Realtime performance

### DISCUSSION THREAD

**Discussion by Antix-Development** [jchristn/SuperSimpleTcp#184] [Q&A] [answered]
I'm quite new to networking and am trying to make an application using SuperSimpleTCP which mirrors keyboard and mouse events from a master computer to one or more connected slave computers when the ALT key is being held.

I have this working but testing has revealed that on the slave computers mousemovement is super laggy.

So I'm wondering if SuperSimpleTCP isn't suitable for my requirements or if there's something I'm not doing properly.

Does anybody have any suggestions to what I might try to speedup my application? Or maybe suggest some other package that would be more suitable? Thanks.

**Comment by jchristn** [comment, marked as answer]
Hi @Antix-Development this library makes extensive use of tasks and events, which can add obvious delays into interactions, especially if you're talking about mouse movements on one machine appearing on another.  I have another library, which is more low-level, called CavemanTcp.  It's mostly for building state machines (e.g. you need to know how many bytes to read from the underlying socket) but may be lower latency and higher throughput for your app.  It's a bit different of an integration model though; it doesn't use events for data that arrives - you have to be waiting for it.

**Comment by Antix-Development** [comment]
@jchristn Thanks! I'll checkout CavemanTCP later on today :)
